rascal-lsp/src/main/rascal/lang/rascal/tests/rename/Annotations.rsc

```diff
@@ -0,0 +1,60 @@
+@license{
+Copyright (c) 2018-2023, NWO-I CWI and Swat.engineering
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice,
+this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+this list of conditions and the following disclaimer in the documentation
+and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+}
+module lang::rascal::tests::rename::Annotations
+
+import lang::rascal::tests::rename::TestUtils;
+import lang::rascal::lsp::refactor::Exception;
+
+test bool annoAtDecl() = {0, 1} == testRenameOccurrences("
+    'x = d();
+    'x@foo = 8;
+    ", decls = "
+    'data D = d();
+    'anno int D@foo;
+");
+
+test bool annoAtUse() = {0, 1} == testRenameOccurrences("
+    'x = d();
+    'int i = x@foo;
+    ", decls = "
+    'data D = d();
+    'anno int D@foo;
+", cursorAtOldNameOccurrence = 1);
```

Comment:
There are many tests in this PR that have equal bodies, except for the value of `cursorAtOldNameOccurrence`. As this value should appear in the result of `testRenameOccurrences` (right?), could the test framework perhaps be extended to check the result for all intended occurrences? This would remove the need for equal tests and make sure all occurrences are tested.

Reply:
Very useful suggestion. I will address this in a future PR.